libsolidity/formal/SMTChecker.cpp

```diff
@@ -853,9 +896,30 @@ void SMTChecker::arithmeticOperation(BinaryOperation const& _op)
 			m_interface->addAssertion(right != 0);
 		}
 
-		checkUnderOverflow(value, intType, _op.location());
+		addOverflowTarget(
+			OverflowTarget::Type::All,
+			_op.annotation().commonType,
+			value,
+			_op.location()
+		);
 
 		defineExpr(_op, value);
```

Comment:
Do we actually need this now, if it's re-defined two lines down anyways?

Reply:
Yes, for the `else` case (no underflow/overflow).

Reply:
Ah no, you're right, there's no guards now. Will remove.